### PR TITLE
Add NICO data-module.

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

File: bolts/data/datamodules/vision/nico.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+"""CelebA DataModule."""
```
Comment (thomkeh) [inline bolts/data/datamodules/vision/nico.py:1]:
```suggestion
"""NICO DataModule."""
```

File: bolts/data/datasets/vision/celeba.py [hunk 2]
```diff
@@ -45,8 +48,8 @@ def __init__(
 
         dataset, self._img_dir = em.celeba(
             download_dir=root,
-            label=superclass,
-            sens_attr=subclass,
+            label=superclass,  # type: ignore
+            sens_attr=subclass,  # type: ignore
```
Comment (thomkeh) [inline bolts/data/datasets/vision/celeba.py:52]: Could use a TypeGuard here, if you wanted to be fancy

File: bolts/fair/data/datamodules/base.py [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
 import logging
-from typing import List, Optional, Sequence, Union
+from typing import Optional, Sequence, Union
 
 from kit import implements
 from kit.torch import StratifiedSampler
 import pytorch_lightning as pl
 from torch.utils.data import DataLoader, Dataset, Sampler
 
+from bolts.data.datamodules import BaseDataModule as _BaseDataModule
```
Comment (thomkeh) [inline bolts/fair/data/datamodules/base.py:10]: Is this right? It looks like a circular import.

File: bolts/fair/data/datamodules/vision/celeba.py [hunk 4]
```diff
@@ -4,26 +4,28 @@
...
-from .base import VisionBaseDataModule
+from .base import BaseVisionDataModule
 
 __all__ = ["CelebaDataModule"]
 
 
-class CelebaDataModule(VisionBaseDataModule):
-    """CelebA Dataset."""
+class CelebaDataModule(BaseVisionDataModule):
+    """Fairness-oriented CelebA data-module."""
```
Comment (thomkeh) [inline bolts/fair/data/datamodules/vision/celeba.py:22]: Eventually this datamodule should use the CelebA dataset class I added, right?

File: bolts/data/datamodules/base.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    _train_data: Dataset
+    _val_data: Dataset
+    _test_data: Dataset
+
+    def __init__(
+        self,
+        batch_size: int,
+        val_prop: Union[float, int],
+        test_prop: Union[float, int],
```
Comment (olliethomas) [inline bolts/data/datamodules/base.py:25]: these can only be floats though? as `train_prop = 1 - val_prop - test_prop`

File: bolts/data/datamodules/common.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+from __future__ import annotations
+
+from typing_extensions import Literal
+
+__all__ = ["Stage"]
+
+Stage = Literal["fit", "validate", "test"]
```
Comment (olliethomas) [inline bolts/data/datamodules/common.py:7]: we use different terminology elsewhere (train, val, test) - we should consolidate

Comment (MylesBartlett) [reply bolts/data/datamodules/common.py:7]: We do, and incorrectly. I checked the lightning docs last night - what I have here is the correct set of options for Stage.

File: bolts/fair/data/datamodules/vision/celeba.py (hunk 4 above)
Comment (MylesBartlett) [reply bolts/fair/data/datamodules/vision/celeba.py:22]: Yes but it's beyond the scope of this PR

File: bolts/fair/data/datamodules/base.py (hunk 3 above)
Comment (MylesBartlett) [reply bolts/fair/data/datamodules/base.py:10]: it's not a circular import - what makes you think so? The fair modules depend on the non-fair ones, never the converse.

Comment (thomkeh) [reply bolts/fair/data/datamodules/base.py:10]: oh, I missed the `.fair` part, sorry

Comment (thomkeh) [reply bolts/fair/data/datamodules/base.py:10]: still, it might not be the best idea to have two identically-named classes in the code base

**Pull request merged**